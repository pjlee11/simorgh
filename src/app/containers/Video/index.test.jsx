import React from 'react';
import {
  shouldShallowMatchSnapshot,
  isNull,
} from '../../helpers/tests/testHelpers';
import {
  blockArrayModel,
  rawVideoModel,
  rawVideoBlock,
  imageBlock,
} from '../../models/blocks';
import VideoContainer from './index';

describe('Video', () => {
  const rVB = rawVideoBlock(
    rawVideoModel('urn:bbc:pips:pid:p064nsyw', 'p064nsz3', 'clip', '299'),
  );

  describe('with no data', () => {
    shouldShallowMatchSnapshot(
      'should not render anything',
      <VideoContainer />,
    );
  });

  describe('with data', () => {
    const videoData = {
      blocks: [
        {
          type: 'rawVideo',
          model: {
            locator: 'urn:bbc:pips:pid:p064nsyw',
            versionID: 'p064nsz3',
            kind: 'clip',
            duration: '299',
          },
        },
        {
          type: 'image',
          model: {
            blocks: [
              {
                type: 'rawImage',
                model: {
                  locator: '/cpsprodpb/5BD5/production/_101690532_2.jpg',
                },
              },
            ],
          },
        },
      ],
    };

    shouldShallowMatchSnapshot(
      'should render the important props in divs',
      <VideoContainer {...videoData} />,
    );
  });

  describe('with data but no image', () => {
    const data = blockArrayModel([rVB, null]);

    shouldShallowMatchSnapshot(
      'should only render the video',
      <VideoContainer {...data} />,
    );
  });

  describe('with data but no raw image', () => {
    const rIB = null;
    const img = imageBlock(rIB);
    const data = blockArrayModel([rVB, img]);

    isNull('should be null', <VideoContainer {...data} />);
  });
});
