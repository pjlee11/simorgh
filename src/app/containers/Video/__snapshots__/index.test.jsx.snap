// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video with data but no image should only render the video 1`] = `null`;

exports[`Video with data should render the important props in divs 1`] = `
<Video
  duration="299"
  imageLocator="/cpsprodpb/5BD5/production/_101690532_2.jpg"
  kind="clip"
  rawImageSrc="https://ichef.bbci.co.uk/news/640/cpsprodpb/5BD5/production/_101690532_2.jpg"
  versionID="p064nsz3"
  videoLocator="urn:bbc:pips:pid:p064nsyw"
/>
`;

exports[`Video with no data should not render anything 1`] = `null`;
